### ISSUE TITLE
BiocManager ignores force=TRUE when installing from github

### ISSUE THREAD

**Issue by lgeistlinger** [Bioconductor/BiocManager#101]
Just wondering:

```
> BiocManager::install("waldronlab/CNVRanger")
'getOption("repos")' replaces Bioconductor standard repositories, see
'?repositories' for details

replacement repositories:
    CRAN: https://cran.rstudio.com/
    BioCsoft: https://bioconductor.org/packages/3.12/bioc
    BioCann: https://bioconductor.org/packages/3.12/data/annotation
    BioCexp: https://bioconductor.org/packages/3.12/data/experiment
    BioCworkflows: https://bioconductor.org/packages/3.12/workflows

Bioconductor version 3.13 (BiocManager 1.30.13), R Under development (unstable)
  (2021-03-24 r80111)
Installing github package(s) 'waldronlab/CNVRanger'
Skipping install of 'CNVRanger' from a github remote, the SHA1 (7badab56) has not changed since last install.
  Use `force = TRUE` to force installation
```

No visible action taken upon addition of `force= TRUE`: 
```
> BiocManager::install("waldronlab/CNVRanger", force = TRUE)
'getOption("repos")' replaces Bioconductor standard repositories, see
'?repositories' for details

replacement repositories:
    CRAN: https://cran.rstudio.com/
    BioCsoft: https://bioconductor.org/packages/3.12/bioc
    BioCann: https://bioconductor.org/packages/3.12/data/annotation
    BioCexp: https://bioconductor.org/packages/3.12/data/experiment
    BioCworkflows: https://bioconductor.org/packages/3.12/workflows

Bioconductor version 3.13 (BiocManager 1.30.13), R Under development (unstable)
  (2021-03-24 r80111)
Installing github package(s) 'waldronlab/CNVRanger'
Skipping install of 'CNVRanger' from a github remote, the SHA1 (7badab56) has not changed since last install.
  Use `force = TRUE` to force installation
```

Is this intended?

**Comment by mtmorgan** [comment]
Can you please try after updating to the  'passForceToRemotes' branch, e.g.,
```
remotes::install_github("Bioconductor/BiocManager", ref = "passForceToRemotes")
```

**Comment by lgeistlinger** [comment]
Thanks, that works after installing from the branch you indicated:

``` 
> BiocManager::install("waldronlab/CNVRanger", force = TRUE)
'getOption("repos")' replaces Bioconductor standard repositories, see
'?repositories' for details

replacement repositories:
    CRAN: https://cran.rstudio.com/
    BioCsoft: https://bioconductor.org/packages/3.12/bioc
    BioCann: https://bioconductor.org/packages/3.12/data/annotation
    BioCexp: https://bioconductor.org/packages/3.12/data/experiment
    BioCworkflows: https://bioconductor.org/packages/3.12/workflows

Bioconductor version 3.13 (BiocManager 1.30.13.3), R Under development
  (unstable) (2021-03-24 r80111)
Installing github package(s) 'waldronlab/CNVRanger'
Downloading GitHub repo waldronlab/CNVRanger@HEAD
Running `R CMD build`...
* checking for file ‘/private/var/folders/q9/zp53ds5x6s9g62l77whflgyc0000gn/T/RtmpA84OR7/remotese5e3755a73cb/waldronlab-CNVRanger-7badab5/DESCRIPTION’ ... OK
* preparing ‘CNVRanger’:
* checking DESCRIPTION meta-information ... OK
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
* building ‘CNVRanger_1.7.3.tar.gz’
* installing *source* package ‘CNVRanger’ ...
** using staged installation
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (CNVRanger)
```

**Comment by mtmorgan** [comment]
Also, it looks like you've manually set `options(repos = ...)` or similar, over-riding the repositories BiocManager recommends with those for BIoconductor version 3.12 -- you've created an invalid installation, with packages from the release branch, tested under R-4.0.5, installed onto the R-4.1 pre-release. This comes pretty close to defeating the purpose of BiocManager...

Re-establish sanity by removing the `repos` customization, removing BiocVersion
```
remove.packages("BiocVersion")  # repeat until no more installations available
```
then starting again
```
BiocManager::install(version = "3.13")
BiocManager::valid()
## BiocManager::install(ask = FALSE) # update all invalid packages
```

**Comment by lgeistlinger** [comment]
All correct, Martin :-)

I am currently experimenting (for the first time) with an RStudio Connect server and was following the RStudio Connect admin guide for package management https://docs.rstudio.com/connect/admin/r/package-management/ that seemed me to explicitely prompt me for setting the repos. That's largely unrelated to the issue here but causes the observations you made due to setting this in my Rprofile ...

local({
  r <- getOption("repos")
  r["CRAN"] <- "https://cran.rstudio.com/"
  r["BioCsoft"] <- "https://bioconductor.org/packages/3.12/bioc"
  r["BioCann"] <- "https://bioconductor.org/packages/3.12/data/annotation"
  r["BioCexp"] <- "https://bioconductor.org/packages/3.12/data/experiment"
  r["BioCworkflows"] <- "https://bioconductor.org/packages/3.12/workflows"
  options(repos = r)
})

**Comment by lgeistlinger** [comment]
(that is because the RSC server uses `packrat` instead of `BiocManager` for package management, and we did not find a way yet to change that ...)

**Comment by LiNk-NY** [comment]
Thanks for reporting! This was resolved in af0aaadf68b92f16e8888f1fe0c3619b3ad9303f
